### PR TITLE
KDESKTOP-659-We-should-never-try-to-upload-a-chunk-with-size-0

### PR DESCRIPTION
**Pull request by ClementKunz**
We should never send a request with a chunk size 0

### CODE REVIEW

File: src/libsyncengine/jobs/network/upload_session/uploadsession.cpp [hunk 1]
```diff
@@ -285,62 +285,72 @@ bool UploadSession::sendChunks() {
...
-                }
+            }
 
-                if (XXH3_64bits_update(state, chunkJob->chunkHash().data(), chunkJob->chunkHash().length()) == XXH_ERROR) {
-                    LOGW_WARN(_logger,
-                              L"Checksum computation " << jobId() << L" failed for file " << Path2WStr(_filePath).c_str());
-                    checksumError = true;
-                    break;
-                }
+            std::streamsize actualChunkSize = file.gcount();
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/jobs/network/upload_session/uploadsession.cpp:290]:
```suggestion
            const std::streamsize actualChunkSize = file.gcount();
```

File: src/libsyncengine/jobs/network/upload_session/uploadsession.cpp [hunk 2]
```diff
@@ -285,62 +285,72 @@ bool UploadSession::sendChunks() {
...
+            std::string chunkContent = std::string(memblock.get(), actualChunkSize);
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/jobs/network/upload_session/uploadsession.cpp:291]: Could this be moved after the conditional `if (actualChunkSize <= 0)`?

File: src/libsyncengine/jobs/network/upload_session/uploadsession.cpp [hunk 3]
```diff
@@ -285,62 +285,72 @@ bool UploadSession::sendChunks() {
...
-                        LOGW_WARN(_logger, L"Failed to upload chunk " << chunkNb << L" of file "
-                                                                      << Path2WStr(_filePath.filename()).c_str());
-                        _exitCode = exitCode;
-                        _exitCause = chunkJob->exitCause();
-                        _jobExecutionError = true;
-                        break;
-                    }
-
-                    _progress += actualChunkSize;
+            std::shared_ptr<UploadSessionChunkJob> chunkJob = nullptr;
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/jobs/network/upload_session/uploadsession.cpp:303]: It seems that a shared pointer is initialized to `nullptr` by default: https://stackoverflow.com/questions/57310392/will-a-default-constructed-empty-shared-ptr-automatically-be-initialized-to-nu.

File: src/libsyncengine/jobs/network/upload_session/uploadsession.cpp [hunk 4]
```diff
@@ -285,62 +285,72 @@ bool UploadSession::sendChunks() {
...
-                    // Set VFS callbacks
-                    chunkJob->setVfsUpdateFetchStatusCallback(_vfsUpdateFetchStatus);
-                    chunkJob->setVfsSetPinStateCallback(_vfsSetPinState);
-                    chunkJob->setVfsForceStatusCallback(_vfsForceStatus);
-                }
+            if (actualChunkSize <= 0) {
+                LOG_ERROR(_logger, "Chunk size is 0");
+#ifdef NDEBUG
+                sentry_capture_event(
+                    sentry_value_new_message_event(SENTRY_LEVEL_WARNING, "Upload chunke error", "Chunk size is 0"));
```
[the remark this answers is not in the document]
Comment (ClementKunz) [reply src/libsyncengine/jobs/network/upload_session/uploadsession.cpp]: Oups... it's fixed

File: src/libsyncengine/jobs/network/upload_session/uploadsession.cpp (hunk 2 above)
Comment (ClementKunz) [reply src/libsyncengine/jobs/network/upload_session/uploadsession.cpp]: Yes, you are right, it's done

**Comment by ClementKunz** [comment]
SonarCloud issue "Replace this use of "bind" with "std::bind_front" cannot be fixed for now because std::bind_front appears with C++20

**Review by ChristopheLarchier — APPROVED** [review]

**Pull request merged**